src/implementations/twig/components/inpage-navigation/inpage-navigation.story.js

```diff
@@ -1,112 +1,127 @@
 import inpageNavigation from './inpage-navigation.html.twig';
 import notes from './README.md';
 
-const demoText = loremIpsum({ count: 25 });
-const cssText =
-  'height: 200px; margin: 0 0 1em; background: #ececec; display:flex; padding: 1em; align-items: center; justify-content: center;';
-const btnLeftLabel = 'Inject a block in the left sidebar';
-const btnMainLabel = 'Inject a generic block in the main column';
-const btnIdLabel =
-  'Inject a new <h2 id="..." in the main column (random order)';
-const btnIdRemoveLabel =
-  'Remove an <h2 id="..." from the main column (random order)';
-// Handlers.
-// Remove from content button.
-const btnIdRemoveHandler = () => {
-  const numH2s = document.querySelectorAll('.ecl-col-l-9 h2[id]').length;
-  const position = Math.floor(Math.random() * Math.floor(numH2s));
-  const randomH2 = document.querySelectorAll('.ecl-col-l-9 h2[id]')[position];
-  randomH2.nextSibling.outerHTML = '';
-  randomH2.outerHTML = '';
-  // Prevent the story from being reloaded.
-  return false;
-};
-// Add generic container in the left sidebar.
-const btnLeftHandler = () => {
-  const btnLeftElement = document.querySelector('.inPageDemoSidebar');
-  const btnLeftTag = document.createElement('div');
-  btnLeftTag.style.cssText = cssText;
-  btnLeftTag.innerHTML =
-    '<span>This is a block injected in the left sidebar</span>';
-  btnLeftElement.append(btnLeftTag);
-  // Prevent the story from being reloaded.
-  return false;
-};
-// Add generic container in the main column.
-const btnMainHandler = () => {
-  const btnMainElement = document.querySelector('.inPageDemoContent');
-  const btnMainTag = document.createElement('div');
-  btnMainTag.style.cssText = cssText;
-  btnMainTag.innerHTML =
-    '<h2>This is an injected block in the Main column</h2>';
-  btnMainElement.append(btnMainTag);
-  // Prevent the story from being reloaded.
-  return false;
+const getArgTypes = (data) => {
+  const argTypes = {};
+  data.links.forEach((item, i) => {
+    argTypes[`heading${i + 1}`] = {
+      name: `heading ${i + 1}`,
+      type: { name: 'string', required: true },
+      defaultValue: item.label,
+      description: `The label for heading  ${i + 1}`,
+      table: {
+        type: { summary: 'string' },
+        defaultValue: { summary: '' },
+        category: 'Content',
+      },
+      control: {
+        type: 'text',
+      },
+    };
+  });
+
+  argTypes.blockLeft = {
+    name: 'left sidebar block',
+    type: { name: 'boolean' },
+    defaultValue: false,
+    description: 'Inject a test content block in the left sidebar',
+    table: {
+      category: 'Test content',
+    },
+    control: {
+      type: 'boolean',
+    },
+  };
+  argTypes.blockMain = {
+    name: 'main content block',
+    type: { name: 'boolean' },
+    defaultValue: false,
+    description: 'Inject a test content block in the main column',
+    table: {
+      category: 'Test content',
+    },
+    control: {
+      type: 'boolean',
+    },
+  };
+
```

Comment:
As it's not possible to have button type controls, I didn't go further on this feature. 
I also don't find it particularly relevant. what do you think?

Reply:
I didn't know the idea behind the buttons knobs injecting markup to demonstrate such a sophisticated behavior.
Feels a bit unnecessary to me, but I don't have a strong argument against it. I spent some time in the docs and the issue queue of storybook, there is no button control and there is no way to create a custom control at the moment. (at least not in a stable way).

I'm personally ok to keep the knobs with the buttons if it's a high importance demo. If we decide to keep the knobs and the buttons, let's please include texts explaning the feature rather than lorem ipsum which doesn't bring value to the demo

Reply:
Well, it's a feature of the component that is otherwise hidden and difficult to discover, but since this is maybe the only use case where we would find some irreplaceable knob i wouldn't really keep the knobs addon in place just for this. Fine to me to give up about them.